backend/src/main/kotlin/code/nebula/cipherquest/advisor/TitleQuestionAnswerAdvisor.kt

```diff
@@ -46,43 +47,40 @@ class TitleQuestionAnswerAdvisor(
         val searchRequestToUse =
             SearchRequest
                 .from(this.searchRequest)
-                .withQuery(request.userText())
-                .withFilterExpression(doGetFilterExpression(context))
+                .query(request.userText())
+                .filterExpression(doGetFilterExpression(context))
+                .build()
 
         // 2. Search for similar documents in the vector store.
         val documents = vectorStore.similaritySearch(searchRequestToUse)
 
         if (documents != null) {
-            context[RETRIEVED_DOCUMENTS] = documents
-        }
-
-        messageContext.context["sources"] =
-            documents
-                .map {
-                    mapOf(
-                        "id" to it.id,
-                        "title" to (
-                            it.metadata["source"]
+            messageContext.sources =
+                documents
+                    .map { doc ->
+                        Source(
+                            doc.id,
+                            doc.metadata["source"]
                                 ?.toString()
                                 ?.split(".")
                                 ?.getOrNull(1)
-                                .orEmpty()
-                        ),
-                    )
-                }.toList()
+                                .orEmpty(),
+                        )
+                    }.toList()
+        }
 
         // 3. Create the context from the documents.
         val documentContext =
             documents
                 ?.joinToString(
                     System.lineSeparator(),
-                ) { document -> "Title: ${document.metadata["source"]}\n${document.content}" }
+                ) { document -> "Title: ${document.metadata["source"]}\n${document.text}" }
```

Comment:
why `text` instead of content?

Reply:
The new `Document` API manages separately `text` and `media` content.